### PR TITLE
Update machine-controller to v1.27.1

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Upate machine-controller to v1.27.1
```

### CODE REVIEW

File: pkg/resources/machinecontroller/webhook.go [hunk 1]
```diff
@@ -53,6 +54,24 @@ var (
...
+			}
+
+			externalCloudProvider := data.Cluster().Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider]
+			if externalCloudProvider {
+				args = append(args, "-node-external-cloud-provider=true")
+			}
+
+			featureGates := data.GetCSIMigrationFeatureGates()
+			if len(featureGates) > 0 {
+				args = append(args, "-node-kubelet-feature-gates", strings.Join(featureGates, ","))
```
Comment (kron4eg) [inline pkg/resources/machinecontroller/webhook.go:72]: If you think this line is missing `"RotateKubeletServerCertificate=true"` then it's not. `"RotateKubeletServerCertificate=true"` defaulting to true is now part of the machine-controller (even when `-node-kubelet-feature-gates` is defined but without `RotateKubeletServerCertificate`.

**Review by xmudrii — CHANGES_REQUESTED** [review]
Can you also update the [machine-controller dependency](https://github.com/kubermatic/kubermatic/blob/24376f2209aca78f52be35d7814ca7d7494c3b23/go.mod#L48), so that:

* we're sure everything works fine with the newer version of the dependency
* we can use the newly added types (such as the annotations)

**Comment by kron4eg** [comment]
> Can you also update the [machine-controller dependency](https://github.com/kubermatic/kubermatic/blob/24376f2209aca78f52be35d7814ca7d7494c3b23/go.mod#L48), so that:
> 
> * we're sure everything works fine with the newer version of the dependency

done

> * we can use the newly added types (such as the annotations)

should be done in the followup PRs done under #6551

**Review by xmudrii — APPROVED** [review]
/lgtm

**Pull request merged**